source/common/http/match_delegate/config.cc

```diff
@@ -279,13 +299,48 @@ Envoy::Http::FilterFactoryCb MatchDelegateConfig::createFilterFactoryFromProtoTy
                                      validation_visitor.errors()[0]));
   }
 
-  Matcher::MatchTreeSharedPtr<Envoy::Http::HttpMatchingData> match_tree = factory_cb();
+  Matcher::MatchTreeSharedPtr<Envoy::Http::HttpMatchingData> match_tree = nullptr;
+
+  if (factory_cb.has_value()) {
+    match_tree = factory_cb.value()();
+  }
+
   return [filter_factory, match_tree](Envoy::Http::FilterChainFactoryCallbacks& callbacks) -> void {
     Factory::DelegatingFactoryCallbacks delegating_callbacks(callbacks, match_tree);
     return filter_factory(delegating_callbacks);
   };
 }
 
+Router::RouteSpecificFilterConfigConstSharedPtr
+MatchDelegateConfig::createRouteSpecificFilterConfigTyped(
+    const envoy::extensions::common::matching::v3::ExtensionWithMatcherPerRoute& proto_config,
+    Server::Configuration::ServerFactoryContext& server_context,
+    ProtobufMessage::ValidationVisitor&) {
+  return std::make_shared<FilterConfigPerRoute>(proto_config, server_context);
+}
+
+Matcher::MatchTreeSharedPtr<Envoy::Http::HttpMatchingData>
+FilterConfigPerRoute::createFilterMatchTree(
+    const envoy::extensions::common::matching::v3::ExtensionWithMatcherPerRoute& proto_config,
+    Server::Configuration::ServerFactoryContext& server_context) {
+  auto requirements =
+      std::make_unique<envoy::extensions::filters::common::dependency::v3::MatchingRequirements>();
+  requirements->mutable_data_input_allow_list()->add_type_url(TypeUtil::descriptorFullNameToTypeUrl(
+      envoy::type::matcher::v3::HttpRequestHeaderMatchInput::descriptor()->full_name()));
+  Server::Configuration::FactoryContext* context =
+      reinterpret_cast<Server::Configuration::FactoryContext*>(&server_context);
+  Envoy::Http::Matching::HttpFilterActionContext action_context{
+      server_context.scope().symbolTable().toString(server_context.scope().prefix()), *context};
```

Comment:
From our last conversation, here is `config_test` only. So we should append the `http.` in front same as logic [here](https://github.com/envoyproxy/envoy/blob/c392328fd179e26fdc8b3b12211baa7e6cf47009/source/extensions/filters/network/http_connection_manager/config.cc#L306), no?